File: SpringBoot/lab4/front/src/app/interface/Absence.ts
import { Student } from "./Student";

export interface Absence {
haveReason: boolean;
id: any;
  reason: string;
  date: string;  // Use string for the date instead of Date, since the backend sends it as a string
  student: Student | number;  // Can be a Student object or the student ID
}

File: SpringBoot/lab4/src/main/java/com/lsi/lab4/entity/Absence.java
package com.lsi.lab4.entity;


import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.util.Date;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class Absence implements Serializable {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Temporal(TemporalType.DATE)
  private Date date;

  private Boolean haveReason = false;

  @Column(nullable = true)
  private String reason;


  @JsonIgnore
  @ManyToOne
  @JoinColumn(name = "student_id")
  private Student student;

  // Getters and setters
}
